core/DataAccess/Model.php

```diff
@@ -354,50 +354,47 @@ public function getArchiveIdsForDeletedSites($archiveTableName)
      * Get a list of IDs of archives with segments that no longer exist in the DB. Excludes temporary archives that 
      * may still be in use, as specified by the $oldestToKeep passed in.
      * @param string $archiveTableName
-     * @param array $segmentHashesById  Whitelist of existing segments, indexed by site ID
+     * @param array $segments  List of segments to match against
      * @param string $oldestToKeep Datetime string
      * @return array With keys idarchive, name, idsite
      */
-    public function getArchiveIdsForDeletedSegments($archiveTableName, array $segmentHashesById, $oldestToKeep)
+    public function getArchiveIdsForSegments($archiveTableName, array $segments, $oldestToKeep)
     {
-        $validSegmentClauses = [];
-
-        foreach ($segmentHashesById as $idSite => $segments) {
-            // segments are md5 hashes and such not a problem re sql injection. for performance etc we don't want to use
-            // bound parameters for the query
-            foreach ($segments as $segment) {
-                if (!preg_match('/^[a-z0-9A-Z]+$/', $segment)) {
-                    throw new Exception($segment . ' expected to be an md5 hash');
-                }
-            }
-
-            // Special case as idsite=0 means the segments are not site-specific
-            if ($idSite === 0) {
-                foreach ($segments as $segmentHash) {
-                    $validSegmentClauses[] = '(name LIKE "done' . $segmentHash . '%")';
-                }
-                continue;
-            }
-
-            $idSite = (int)$idSite;
-
-            // Vanilla case - segments that are valid for a single site only
-            $sql = '(idsite = ' . $idSite . ' AND (';
-            $sql .= 'name LIKE "done' . implode('%" OR name LIKE "done', $segments) . '%"';
-            $sql .= '))';
-            $validSegmentClauses[] = $sql;
+        $segmentClauses = [];
+        foreach ($segments as $segment) {
+            $segmentClauses[] = self::getDeletedSegmentWhereClause($segment);
```

Comment:
btw be good to add the segmentClause only if
```
if(!empty($segment['definition'])) {
     $segmentClauses[] = self::getDeletedSegmentWhereClause($segment);
 }
```

because otherwise it would delete archives where `done%`. It shouldn't ever happen but could to have a safety net here just in case in the future somewhere `definition` suddenly gets lost or someone is not using the method correctly.

Reply:
Actually this will be really important. I just had triggered a case where `definition` wasn't set for some reason. So we need to make sure to not accidentally delete regular archives

Reply:
@katebutler be good to add the definition check just to be 100% safe.